tests/test_browser.py

```diff
@@ -56,7 +56,33 @@ def shell_with_script(shell_file, output_file, replacement):
     with open(output_file, 'w') as output:
       output.write(input.read().replace('{{{ SCRIPT }}}', replacement))
 
-requires_hardware = unittest.skipIf(os.environ.get("EM_LACKS_HARDWARE_ACCESS"), "This test requires hardware access including graphics and sound devices")
+def requires_hardware(hardware):
```

Comment:
I'm still unsure of the benefits of this. Let me explain in more detail what I was proposing before, maybe I wasn't clear: first, we'd define
```
requires_graphics_hardware = unittest.skipIf(os.environ.get("EM_LACKS_GRAPHICS_HARDWARE"), "This test requires graphics hardware")
requires_sound_hardware = unittest.skipIf(os.environ.get("EM_LACKS_SOUND_HARDWARE"), "This test requires sound hardware")
```
and then the tests would use `@requires_graphics_hardware` or `@requires_sound_hardware`.

I think this is preferable as it's just 2 simple lines to define the python decorators, and also, it avoids using a python string when using the decorators (which has the risk of a typo). What do you think?

(There is no need to support `EM_LACKS_HARDWARE_ACCESS` in a backwards-compatible manner.)